packages/api/review/teamsfx-api.api.md

```diff
@@ -682,6 +682,8 @@ export interface ProjectSettings {
     // (undocumented)
     appName: string;
     // (undocumented)
+    programmingLanguage?: string;
```

Comment:
Nit:  one thing comes to my mind is, how about extract this programming language as enum, and all stakeholders share the same standard of the programming language?
Currently, bot plugin will validate this programming language to check if it's in list of ['javascript', 'typescript']. If the enum definition is in a central place, it would be better.

Reply:
Making the `programmingLanguage` as enum sounds good to me.
As the purpose of this PR is to move this setting from `env.default.json` to `settings.json`, I would prefer keep the logic of consuming the `programmingLanguage` not change too much.

Since mainly bot & function plugin use the `progammingLanguage` setting, as the plugin owners, you can add a enum (e,g. in `api` package) to facilitate using this setting.